### PR TITLE
feat(reporter): Add `parallelIndex` field to `TestResult`

### LINKED ISSUE #19541: [Feature] Make `parallelIndex` available in public API for reporters
Sadly https://playwright.dev/docs/api/class-workerinfo#worker-info-parallel-index is not available in `TestResult` object for 
reporters. Let's make it visible.

Successor of https://github.com/microsoft/playwright/issues/19519

### PR DESCRIPTION
**Pull request by vovsemenv**
fixes https://github.com/microsoft/playwright/issues/19541

### CODE REVIEW

**Comment by mxschmitt** [comment]
6 bots are failing

**Comment by vovsemenv** [comment]
@mxschmitt fixed

File: docs/src/test-reporter-api/class-testresult.md [hunk 1]
```diff
@@ -78,3 +78,9 @@ List of steps inside this test run.
 Index of the worker where the test was run. If the test was not run a single time, for example when the user interrupted testing, the only result will have a `workerIndex` equal to `-1`.
 
 Learn more about [parallelism and sharding](../test-parallel.md) with Playwright Test.
+
+## property: TestResult.parallelIndex
+* since: v1.30
+- type: <[int]>
+
+The index of the worker between `0` and `workers - 1`. It is guaranteed that workers running at the same time have a different `parallelIndex`. When a worker is restarted, for example after a failure, the new worker process has the same `parallelIndex`.
```
Comment (pavelfeldman) [inline docs/src/test-reporter-api/class-testresult.md:86]: > When a worker is restarted, for example after a failure, the new worker process has the same `parallelIndex`.

This part is not accurate, so I'd suggest dropping it.

File: packages/playwright-test/src/dispatcher.ts [hunk 2]
```diff
@@ -495,7 +496,7 @@ let lastWorkerIndex = 0;
 class Worker extends EventEmitter {
   private process: child_process.ChildProcess;
   private _hash: string;
-  private parallelIndex: number;
+  public parallelIndex: number;
```
Comment (pavelfeldman) [inline packages/playwright-test/src/dispatcher.ts:499]: can this also be readonly to be consistent with workerIndex?

**Review by pavelfeldman — COMMENTED** [review]
Looks good, a couple of small updates and it is good to go.

File: packages/playwright-test/src/dispatcher.ts (hunk 2 above)
Comment (vovsemenv) [reply packages/playwright-test/src/dispatcher.ts]: done

File: docs/src/test-reporter-api/class-testresult.md (hunk 1 above)
Comment (vovsemenv) [reply docs/src/test-reporter-api/class-testresult.md]: done

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**